<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1284</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PicPro</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/images/images/icon.png</normaloff>:/images/images/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QScrollArea" name="scrollArea1">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>10</y>
      <width>250</width>
      <height>250</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents1">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>248</width>
       <height>248</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="showPic1">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QLabel" name="quantificat_label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>430</y>
      <width>60</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>量化等级</string>
    </property>
   </widget>
   <widget class="QComboBox" name="quantificat">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>430</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>原图</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>256</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>64</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>16</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>2</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="sampling">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>390</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>原图</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>512</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>256</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>128</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>64</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>32</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>16</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="sampling_label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>390</y>
      <width>60</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>采样点数</string>
    </property>
   </widget>
   <widget class="QPushButton" name="eight">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>50</y>
      <width>271</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>点击生成8幅位平面表示的二值图</string>
    </property>
   </widget>
   <widget class="QCustomPlot" name="grayhistogram2" native="true">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>300</y>
      <width>361</width>
      <height>181</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="information2">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>490</y>
      <width>251</width>
      <height>91</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;灰度直方图信息&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;像素总数：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;平均灰度：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;中值灰度：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;标准差：&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="threshold">
    <property name="geometry">
     <rect>
      <x>1210</x>
      <y>600</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="value">
     <number>128</number>
    </property>
   </widget>
   <widget class="QLabel" name="threshold_label">
    <property name="geometry">
     <rect>
      <x>985</x>
      <y>600</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>设置阈值，实时显示对应二值图</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea2">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>300</y>
      <width>600</width>
      <height>600</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>598</width>
       <height>598</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="showPic2">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QLabel" name="label_origin">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>10</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>AcadEref</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>原图</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea3">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>650</y>
      <width>250</width>
      <height>250</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>248</width>
       <height>248</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="showPic3">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QCustomPlot" name="grayhistogram1" native="true">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>10</y>
      <width>361</width>
      <height>181</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="information1">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>200</y>
      <width>251</width>
      <height>91</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;灰度直方图信息&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;像素总数：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;平均灰度：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;中值灰度：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;标准差：&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_processed">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>340</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>AcadEref</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>处理后图片</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_pointop">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>470</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>点运算</string>
    </property>
   </widget>
   <widget class="QComboBox" name="pointop">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>470</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>原图</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>负片变换</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>对数变换</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>伽马变换</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>对比度拉伸</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="equilibrium">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>530</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>传统直方图均衡</string>
    </property>
   </widget>
   <widget class="QPushButton" name="equilibriumBBHE">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>530</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>BBHE直方图均衡</string>
    </property>
   </widget>
   <widget class="QLabel" name="rotate_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>600</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>旋转</string>
    </property>
   </widget>
   <widget class="QSlider" name="rotate">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>600</y>
      <width>191</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="degree">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>600</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>0度   </string>
    </property>
   </widget>
   <widget class="QLabel" name="label_scale">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>690</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>缩放  </string>
    </property>
   </widget>
   <widget class="QLabel" name="label_insert">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>650</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>灰度插值法  </string>
    </property>
   </widget>
   <widget class="QComboBox" name="insert">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>650</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>最近邻插值</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>双线性插值</string>
     </property>
    </item>
   </widget>
   <widget class="QDoubleSpinBox" name="scale">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>690</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="prefix">
     <string/>
    </property>
    <property name="suffix">
     <string notr="true"/>
    </property>
    <property name="decimals">
     <number>1</number>
    </property>
    <property name="minimum">
     <double>50.000000000000000</double>
    </property>
    <property name="maximum">
     <double>200.000000000000000</double>
    </property>
    <property name="value">
     <double>100.000000000000000</double>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalmove">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>740</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_horizontalmove">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>740</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>水平移动</string>
    </property>
   </widget>
   <widget class="QSlider" name="verticalmove">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>770</y>
      <width>22</width>
      <height>121</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_verticalmove">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>810</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>垂直移动</string>
    </property>
   </widget>
   <widget class="QPushButton" name="minus">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>110</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>图像相减</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bmp2txt">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>110</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>bmp2txt</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pedestrain">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>110</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>行人分析</string>
    </property>
   </widget>
   <widget class="QPushButton" name="transform">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>170</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>照片变形</string>
    </property>
   </widget>
   <widget class="QPushButton" name="chromatic">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>230</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>彩色图像处理</string>
    </property>
   </widget>
   <widget class="QPushButton" name="convert">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>170</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>图像变换</string>
    </property>
   </widget>
   <widget class="QPushButton" name="enhance">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>170</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>图像增强</string>
    </property>
   </widget>
   <widget class="QPushButton" name="segmentation">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>230</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>图像分割</string>
    </property>
   </widget>
   <widget class="QPushButton" name="compress">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>230</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>图像编码与压缩</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1284</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenPic"/>
   <addaction name="actionSavePic"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpenPic">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/images/images/Open.png</normaloff>:/images/images/Open.png</iconset>
   </property>
   <property name="text">
    <string>打开图片</string>
   </property>
   <property name="toolTip">
    <string>打开图片</string>
   </property>
  </action>
  <action name="actionSavePic">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/images/images/Save.png</normaloff>:/images/images/Save.png</iconset>
   </property>
   <property name="text">
    <string>保存图片</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
